### PR TITLE
Fixed error type implementation for errors that are defined in modules which provide errorDescription in the application. 

### LINKED ISSUE #6541: Fix error type implementation for those errors that are defined in modules (i.e. CocoaPod) and their localized strings are provided in the app bundle
Related to #6498.

eg. `SendInputErrors`, make the error conform to `Error` instead of `LocalizedError` in the module and only conform to `LocalizedError` in the app, i.e. change from this:

```
//module
struct E1: LocalizedError {
}

//app
extension E1 {
    var errorDescription: String? {
      return "E1's localized errorDescription" //using R.string.localizable.xxx()
    }
}
```

to:

```
//module
struct E1: Error {
}

//app
extension E1: LocalizedError {
    var errorDescription: String? {
      return "E1's localized errorDescription" //using R.string.localizable.xxx()
    }
}
```

### PR DESCRIPTION
**Pull request by eviltofu**
Closes #6541

I ran `grep -nriI "LocalizedError" . | grep -i ".swift" | grep -i "module" | sort > ~/Desktop/LocalizedErrors.txt` on the root directory of the repo to find all the instances where `LocalizedError ` occurred in swift files in the module directory. This gave me a list of 19 instances.
```text
./modules/AlphaWalletENS/AlphaWalletENS/ENSError.swift:9:struct ENSError: LocalizedError {
./modules/AlphaWalletFoundation/AlphaWalletFoundation/EtherClient/CastError.swift:5:public struct CastError<ExpectedType>: LocalizedError {
./modules/AlphaWalletFoundation/AlphaWalletFoundation/Ethereum/ABI/ABIError.swift:9:public enum ABIError: LocalizedError {
./modules/AlphaWalletFoundation/AlphaWalletFoundation/KeyManagement/EtherKeystore.swift:11:public enum EtherKeystoreError: LocalizedError {
./modules/AlphaWalletFoundation/AlphaWalletFoundation/KeyManagement/LegacyFileBasedKeystore.swift:9:public enum FileBasedKeystoreError: LocalizedError {
./modules/AlphaWalletFoundation/AlphaWalletFoundation/KeyManagement/SecureEnclave.swift:8:    enum Error: LocalizedError {
./modules/AlphaWalletFoundation/AlphaWalletFoundation/NFT/Enjin/EnjinUserManagementInterceptor.swift:102:    public enum JSONResponseParsingError: Error, LocalizedError {
=> ./modules/AlphaWalletFoundation/AlphaWalletFoundation/Settings/Models/AddCustomChain.swift:6:public enum AddCustomChainError: LocalizedError {
=> ./modules/AlphaWalletFoundation/AlphaWalletFoundation/Settings/Types/SendInputErrors.swift:5:public enum SendInputErrors: LocalizedError {
./modules/AlphaWalletFoundation/AlphaWalletFoundation/SwapToken/NativeSwap/Types/SwapError.swift:10:public enum SwapError: LocalizedError {
./modules/AlphaWalletFoundation/AlphaWalletFoundation/TokenScriptClient/Models/FunctionOrigin.swift:8:public enum FunctionError: LocalizedError {
=> ./modules/AlphaWalletFoundation/AlphaWalletFoundation/TokenScriptClient/Models/XMLHandler.swift:97:    public enum SignatureValidationError: LocalizedError {
./modules/AlphaWalletFoundation/AlphaWalletFoundation/TokenScriptClient/SchemaCheckError.swift:10:public struct SchemaCheckError: LocalizedError {
./modules/AlphaWalletFoundation/AlphaWalletFoundation/Tokens/Logic/GetBlockTimestamp.swift:53:extension JSONRPCKit.JSONRPCError: LocalizedError {
./modules/AlphaWalletFoundation/AlphaWalletFoundation/Tokens/TokenImageFetcher.swift:76:    enum ImageAvailabilityError: LocalizedError {
./modules/AlphaWalletFoundation/AlphaWalletFoundation/Tokens/Types/Web3Error.swift:5:public struct Web3Error: LocalizedError {
./modules/AlphaWalletFoundation/AlphaWalletFoundation/Transfer/TransactionConfigurator.swift:17:public enum TransactionConfiguratorError: LocalizedError {
./modules/AlphaWalletFoundation/AlphaWalletFoundation/Types/DevelopmentForcedError.swift:5:public struct DevelopmentForcedError: LocalizedError {
./modules/AlphaWalletFoundation/AlphaWalletFoundation/Types/RpcJson/JsonRpcError.swift:38:extension JsonRpcError: LocalizedError {
```
I went through all of them and found only two instances which followed your pattern of originally being declared a `LocalizedError` and having an extension in `AlphaWallet` that specifies the `errorDescription`.

Once approved, I will squish the commits into one to merge.

Do I need to go through the `Pods` directory? I skipped it since the code is not under our control.

### CODE REVIEW

**Review by hboon — CHANGES_REQUESTED** [review]
About the PR title/commit message: this is a fix, not a refactoring. This is because before the PR, the app doesn't display the intended, localized string messages for these errors. Would you help to modify them?

Need to fix `TokenLevelTokenScriptDisplayStatus.SignatureValidationError` too

**Comment by eviltofu** [comment]
> About the PR title/commit message: this is a fix, not a refactoring. This is because before the PR, the app doesn't display the intended, localized string messages for these errors. Would you help to modify them?
> 
> Need to fix `TokenLevelTokenScriptDisplayStatus.SignatureValidationError` too

Ack! I checked for `extension SignatureValidationError` and missed this. I will go through the entire list again.

**Comment by hboon** [comment]
> Do I need to go through the Pods directory? I skipped it since the code is not under our control.

Sorry, miss this. You are right. Also we can't modify code in there. If we wanted to we'd have to submit PRs. But I don't think we'll ever need to for error handling since we'd want to wrap any 3rd party ones with our own errors types.

**Comment by eviltofu** [comment]
https://github.com/AlphaWallet/alpha-wallet-ios/blob/bd527e86cca38afecf3846dc74fff43c8f77fe78/modules/AlphaWalletFoundation/AlphaWalletFoundation/Tokens/Logic/GetBlockTimestamp.swift#L53 is one where the pod has the error and our module has the extension.

**Comment by eviltofu** [comment]
`SignatureValidationError` added. @hboon Once approved, I'll squish and merge.

**Comment by hboon** [comment]
@eviltofu thanks. Looks good. Can you squash and push again, then I approve it?

For future PRs, can you squash first, unless it's a biggish PR that really can do with separate commits for easier reviewing? The last time we approved then squash, we forgot the squashing and merge right away.

**Comment by hboon** [comment]
(squash first, and the reviewer can approve and optionally merge right away)

**Comment by eviltofu** [comment]
> @eviltofu thanks. Looks good. Can you squash and push again, then I approve it?
> 
> For future PRs, can you squash first, unless it's a biggish PR that really can do with separate commits for easier reviewing? The last time we approved then squash, we forgot the squashing and merge right away.

I did the separate commits because I didn't know in advance how many entries I needed to correct.

**Review by hboon — APPROVED** [review]

**Pull request merged**
